java/arcs/core/entity/Handle.kt

```diff
@@ -25,12 +25,19 @@ interface Handle {
     suspend fun close()
 }
 
-interface ReadableHandle<T> : Handle {
-    suspend fun onUpdate(action: (T) -> Unit)
+interface ReadableHandle<UpdateType, E : Entity> : Handle {
+    suspend fun onUpdate(action: (UpdateType) -> Unit)
+
+    /**
+     * Creates and returns a [Reference] to the given entity.
+     *
+     * The entity must already be stored and present in the handle before calling this method.
+     */
+    suspend fun createReference(entity: E): Reference<E>
```

Comment:
I'm confused by this, is the developer supposed to always call this method manually? It doesn't seem very ergonomic and I'm hoping it can be hidden completely by the code generator.  What I mean is, if a field is declared as an entity reference, why can't a setFoo() method be generated (or a Set<> implementation) such that it invokes createReference() for you, and then stashes the reference in the field?

I'd expect developer facing code to look like this:
```
manager.employees.add(employee) // employees is a collection of references, employees field is a Set<> 
```

Reply:
Yep, the developer will have to do it themselves unfortunately (at least for now). We need to know which handle the entity is from. One way of doing it would be for `handle.store(entity)` to modify the entity to include information about the handle that it's now stored in (e.g. add storage key). This has issues though: what if you want to store the entity into two different handles?

There's a lot to work through here, we'll have to come back and reconsider it all properly. For now, I think `handle.createReference(entity)` isn't too bad.